scripts/ont_send_loading_info_to_db.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env python
+
+from __future__ import division
+from argparse import ArgumentParser
+from genologics.lims import Lims
+from genologics.config import BASEURI, USERNAME, PASSWORD
+from genologics.entities import Process
+from ont_send_reloading_info_to_db import get_ONT_db
+import sys
+import pandas as pd
+from io import StringIO
+from datetime import datetime as dt
+import re
+from utils import udf_tools
+from ont_generate_samplesheet import make_samplesheet
+import os
+
+DESC = """ Script for EPP "ont_send_loading_info_to_db".
+
+Ensure all samples in the step correspond to an ONT run that was started with the correct samplesheet.
+"""
+
+
+def get_file(lims, currentStep, file_name):
+    file_art = [op for op in currentStep.all_outputs() if op.name == file_name][0]
+    readable = StringIO(lims.get_file_contents(uri=file_art.files[0].uri))
+    return readable
+
+
+def match_to_db_using_run_id(lims, args):
+    currentStep = Process(lims, id=args.pid)
+    timestamp = dt.now().strftime("%y%m%d_%H%M%S")
+
+    arts = [art for art in currentStep.all_outputs() if art.type == "Analyte"]
+
+    db = get_ONT_db()
+    view = db.view("info/all_stats")
+
+    runtime_log = ["Verify the run ID is correct, i.e. visible in GenStat."]
+
+    errors = False
+    for art in arts:
+        try:
+            run_id = art.udf["ONT run name"]
+        except KeyError:
+            runtime_log.append(f"No run name supplied for {art.name}")
+            errors = True
+
+        matching_docs = []
+        for doc in view.rows:
+            if run_id == doc.key:
+                matching_docs.append(doc)
+
+        try:
+            if len(matching_docs) == 0:
+                runtime_log.append(f"{run_id} was not found in the database.")
+                raise AssertionError()
+
+            elif len(matching_docs) > 1:
+                runtime_log.append(
+                    f"{run_id} was found in multiple instances in the database. Contact a database administrator."
+                )
+                raise AssertionError()
+
+            doc_id = matching_docs[0].id
+            doc = db[doc_id]
+
+            dict_to_add = {
+                "step_name": currentStep.type.name,
+                "pid": currentStep.id,
+                "timestamp": timestamp,
+                "qc": art.udf["ONT flow cell QC pore count"],
+                "load_fmol": art.udf["ONT flow cell loading amount (fmol)"],
+            }
+
+            try:
+                try:
+                    doc["lims"]["loading"].append(dict_to_add)
+                except KeyError:
+                    doc["lims"]["loading"] = []
+                    doc["lims"]["loading"].append(dict_to_add)
+            except:
```

Comment:
I think a 
```python
if "lims" not in doc:
    doc["lims"] = {}
``` 
Before this section would be slightly nicer?

Reply:
[2131d7b](https://github.com/SciLifeLab/scilifelab_epps/pull/250/commits/2131d7bf409897642ec1ad486830494dd3a3f153)